### DISCUSSION TITLE
blog/2025-02-month-summary

### DISCUSSION THREAD

**Discussion by giscus** [bbruceyuan/bbruceyuan.github.io#29] [Announcements]
# blog/2025-02-month-summary

回首每一天，似乎都有挺多的遗憾，因为想要做的事情太多了，但是当月底回望这一个月，我觉得很满意了，我想这一次应该是不一样的开始。

https://bruceyuan.com/blog/2025-02-month-summary.html

**Comment by SoupEggQAQ** [comment]
加油博主！看你的视频真的学习了好多！！！

**Comment by Edrex1128** [comment]
哥加油呀！我从b站到微信再到github来找你了（笑哭），我看了你从零手写GPT那节课后有个小问题，本人刚开始学NLP，之前做CV的，可能问题有点初级（笑哭）
（分割线）----------------------------------------
logits = logits.view(batch * seq_len, vocab_size)
targets = targets.view(batch * seq_len)
loss = F.cross_entropy(logits, targets)
这里为什么是用整个logits计算损失，而不是像generate函数里那样把最后一行取出来计算损失？

**Reply by bbruceyuan** [reply]
每个 Token 都算loss不是很合理的选择吗、你可以只算最后一个（但是不是大概率效果不好吗，而且也不合理，但是你可以这么设置）

**Reply by Edrex1128** [reply]
Roger! 谢谢哥
